ios/RCTPSPDFKit/RCTPSPDFKitView.m

```diff
@@ -203,6 +203,37 @@ - (void)addAnnotation:(id)jsonAnnotation {
   }
 }
 
+- (void)removeAnnotation:(id)jsonAnnotation {
+  NSData *data;
+  if ([jsonAnnotation isKindOfClass:NSString.class]) {
+    data = [jsonAnnotation dataUsingEncoding:NSUTF8StringEncoding];
+  } else if ([jsonAnnotation isKindOfClass:NSDictionary.class])  {
+    data = [NSJSONSerialization dataWithJSONObject:jsonAnnotation options:0 error:nil];
+  } else {
+    NSLog(@"Invalid JSON Annotation.");
+    return;
+  }
+
+  PSPDFDocument *document = self.pdfController.document;
+  PSPDFDocumentProvider *documentProvider = document.documentProviders.firstObject;
+
+  BOOL success = NO;
+  if (data) {
+    PSPDFAnnotation *annotationToRemove = [PSPDFAnnotation annotationFromInstantJSON:data documentProvider:documentProvider error:NULL];
```

Comment:
Isn't actually parsing the instant json unnecessary overhead when you can just extract the two relevant properties from the JSON?

Reply:
I think we should work with our Instant JSON API instead of manually parsing the JSON payload. I don't think creating an annotation object like this has a huge performance impact.